File: src/my_py_pkg/my_py_pkg/add_two_ints_client.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts
from functools import partial

class AddTwoIntsClient(Node):
    def __init__(self):
        super().__init__("add_two_ints_client")
        self.client = self.create_client(AddTwoInts, "add_two_ints")

    def call_add_two_ints(self, a, b):
        while not self.client.wait_for_service(1.0):
            self.get_logger().warn("Waiting for Add Two Ints server...")

        request = AddTwoInts.Request()
        request.a = a
        request.b = b

        future = self.client.call_async(request)
        future.add_done_callback(
            partial(self.callback_call_add_two_ints, request=request)
        )

    def callback_call_add_two_ints(self, future, request):
        response = future.result()
        self.get_logger().info(
            f"{request.a} + {request.b} = {response.sum}"
        )

def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClient()
    node.call_add_two_ints(2, 7)
    node.call_add_two_ints(1, 4)
    node.call_add_two_ints(10, 20)
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using namespace std::chrono_literals;
using namespace std::placeholders;

class AddTwoIntsClientNode : public rclcpp::Node
{
public:
  AddTwoIntsClientNode() : Node("add_two_ints_client")
  {
    client_ = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
  }

  void callAddTwoInts(int a, int b)
  {
    while (!client_->wait_for_service(1s))
    {
      RCLCPP_WARN(this->get_logger(), "Waiting for the server...");
    }

    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = a;
    request->b = b;

    client_->async_send_request(
      request,
      std::bind(&AddTwoIntsClientNode::callbackCallAddTwoInts, this, _1)
    );
  }

private:
  void callbackCallAddTwoInts(
    rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedFuture future)
  {
    auto response = future.get();
    RCLCPP_INFO(this->get_logger(), "Sum: %d", (int)response->sum);
  }

  rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedPtr client_;
};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<AddTwoIntsClientNode>();
  node->callAddTwoInts(10, 5);
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
